src/arch/x86/object/vcpu.c

```diff
@@ -290,7 +290,14 @@ static bool_t BOOT_CODE init_vtx_fixed_values(bool_t useTrueMsrs)
     entry_control_high = x86_rdmsr_low(entry_ctls);
     entry_control_low = x86_rdmsr_high(entry_ctls);
 
-    cr0_high = x86_rdmsr_low(IA32_VMX_CR0_FIXED0_MSR);
+    // Intel SDM, Vol. 3C, 24.8:
+    //
+    // > Later processors support a VM-execution control called "unrestricted
+    // > guest" (see Section 25.6.2). If this control is 1, CR0.PE and CR0.PG
+    // > may be 0 in VMX non-root operation (even if the capability MSR
+    // > IA32_VMX_CR0_FIXED0 reports otherwise).1 Such processors allow guest
+    // > software to run in unpaged protected mode or in real-address mode.
+    cr0_high = x86_rdmsr_low(IA32_VMX_CR0_FIXED0_MSR) & ~0x80000001;
```

Comment:
You should check that unrestricted guest mode is actually allowed before doing this.

Also, if there are no defines for individual CR0 bits yet, then please add them and use them here. Some are defined in include/arch/x86/arch/machine/cpu_registers.h already. Same for the IA32_VMX_TRUE_PROCBASED_CTLS_MSR/IA32_VMX_PROCBASED_CTLS_MSR bits you need to check.